rdsaliba/notorious-eng
# F12 Associating a model to an Asset Type

This pull request integrates the feature F10, which allows the user to choose which RUL model to choose for a specific Asset Type. Here are the main points of the PR:
0) SQL scripts
- Make sure the latest dump file has been run (v13)
- Run the model_evaluation script on top of that (in folder v14)
1) Try feature
To try out the feature, go to the model tab in Asset Type info view. From there, you will be able to scroll horizontally through the list of models, see their RMSE value, select the model by clicking on the box as well as saving the association by clicking on save. Saving the feature will update the association in the database as well as set the trained_model to be retrained and the assets associated with that asset type to be updated.
2) Evaluation of models
This pull request also includes the evaluation of single models and not necessarily all of them. A button will be available for each model. Clicking on the button will generate the evaluation and update the corresponding RMSE value/values
3) Input Validation
- Models cannot be evaluated as long as the train slider value or test slider value is 0
- Save button cannot be clicked if no model is selected.
4) Verification of the feature
When saving, the model_id should be properly set in the trained_model table for the corresponding asset type. The retrain tag should also be set to true. Finally, the updated tag of all live assets of that asset type should be set to true.
5) UI
The UI looks relatively bad. This was left on purpose to be done by the UI Refactoring that is currently performed by Jeff and avoid duplicate work.
6) REFACTORING
there are 2 refactoring commits in this pull request. These refactoring types are "extract methods"
